### PR TITLE
feat(notifier): Add makeNotifierFromSubscriber

### LINKED ISSUE #5413: notifier returned by makeNotifierFromAsyncIterable is lossless
## Describe the bug
The design of Notifier is that if you call `getUpdateSince()` without any arguments, you get the latest state.

This is different for the alleged Notifier returned by `makeNotifierFromAsyncIterable`. It never skips elements of the iteration of the underlying subscription.

## Expected behavior

A `Notifier` skips states consistently. Either `makeNotifierFromAsyncIterable` conforms or is renamed. (Or replaced if there's something better.)

### PR DESCRIPTION
**Pull request by gibson042**
Fixes #5413

## Description

Add `makeNotifierFromSubscriber` for lossy consumption of PublishKit subscriber objects via `subscribeLatest` (each `getUpdateSince()` call returns a unique already-settled promise for the latest result, each `getUpdateSince(lastUpdateCount)` call requests the next value [which under the covers constructs a new iterator] and returns a unique promise for it).

Also deprecates `makeNotifierFromAsyncIterable`.

This PR should land with a rebase rather than a squash; I have intentionally included the changes from #5695 along with an immediate reversion for posterity.

### Security Considerations

None known.

### Documentation Considerations

https://github.com/Agoric/documentation/issues/684

### Testing Considerations

More tests to come before merging.

### CODE REVIEW

File: packages/notifier/src/notifier.js
Comment (turadg) [inline packages/notifier/src/notifier.js]: https://www.typescriptlang.org/play#example/jsdoc-deprecated
```suggestion
 * @deprecated adaptor from async iterable to notifier.
```

File: packages/notifier/test/test-notifier-adaptor.js
Comment (turadg) [inline packages/notifier/test/test-notifier-adaptor.js]: please separate this into its own file. that will make it easier when doing watch tests for user to focus on what they want.

File: packages/notifier/test/test-notifier-adaptor.js
Comment (turadg) [inline packages/notifier/test/test-notifier-adaptor.js]: In general,
```suggestion
test.todo('makeNotifierFromSubscriber(fails)')
```
But this will be done before merging?

File: packages/notifier/test/test-notifier-adaptor.js
Comment (turadg) [inline packages/notifier/test/test-notifier-adaptor.js]: please address the floating promises on 250 and 252

File: packages/notifier/test/test-notifier-adaptor.js
Comment (turadg) [inline packages/notifier/test/test-notifier-adaptor.js]: why go through E proxy?

File: packages/notifier/test/test-notifier-adaptor.js
Comment (turadg) [inline packages/notifier/test/test-notifier-adaptor.js]: this sounds like it's making a PublicKit but it's returning something different.

```suggestion
const makeGeometricPublishHelper = ({ conclusionMethod, conclusionValue }) => {
```
Looks like calls have the same arguments. Why parameterize?
```suggestion
const makeGeometricPublishHelper = () => {
```
Also looks like all tests share this first line. Once this is in its own file, it would work well to put in a hook:
```
test.beforeEach(t => {
  Object.assign(t.context, makeGeometricPublishHelper());
}
```
Or even,
```
test.beforeEach(t => {
  Object.assign(t.context, makeGeometricPublishHelper());
  t.context.notifier = await makeNotifierFromSubscriber(t.context.subscriber);
}
```

File: packages/notifier/src/notifier.js
Comment (turadg) [reply packages/notifier/src/notifier.js]: Also please explain here why it's deprecated.

> I have intentionally included the changes from https://github.com/Agoric/agoric-sdk/pull/5695 along with an immediate reversion for posterity.

I request that those two commits be omitted (or this be landed as squash). For master it's noise. If there's something important to convey about how we got to this point then just write it in HEAD. If you want there to be some record of the other work done, it's already in the referenced PR.

**Review by turadg — CHANGES_REQUESTED** [review]

File: packages/notifier/src/notifier.js
Comment (turadg) [inline packages/notifier/src/notifier.js]:
```suggestion
export const makeNotifierFromSubscriber = async subscriber => {
```
I assume `P` is for "Promise" but that's not exactly what this is.

File: packages/notifier/src/notifier.js
Comment (gibson042) [reply packages/notifier/src/notifier.js]: I'm anticipating the attempt to come up again, and would like the prior record to outlive that dead branch.

File: packages/notifier/test/test-notifier-adaptor.js
Comment (gibson042) [reply packages/notifier/test/test-notifier-adaptor.js]: Yes, done now.

File: packages/notifier/test/test-notifier-adaptor.js
Comment (gibson042) [reply packages/notifier/test/test-notifier-adaptor.js]: Address in what way?

File: packages/notifier/test/test-notifier-adaptor.js
Comment (gibson042) [reply packages/notifier/test/test-notifier-adaptor.js]: Abandoned.

File: packages/notifier/test/test-notifier-adaptor.js
Comment (gibson042) [reply packages/notifier/test/test-notifier-adaptor.js]: Renamed for clarity, but kept it outside the test context.

File: packages/notifier/src/notifier.js
Comment (gibson042) [reply packages/notifier/src/notifier.js]: It's an ERef, which _may_ be a promise. I think we do use this pattern to highlight that potential.

**Comment by gibson042** [comment]
@turadg https://github.com/Agoric/agoric-sdk/pull/5737/commits/f40b7ed073e97d961b92f5e28c03a747c91f47e7 addresses everything except the ["floating promises" concern](https://github.com/Agoric/agoric-sdk/pull/5737#discussion_r916970622)... what kind of change are you expecting for that?

File: packages/notifier/test/test-notifier-adaptor.js
Comment (turadg) [reply packages/notifier/test/test-notifier-adaptor.js]: make the warnings go away :) 

either by awaiting or putting `void ` to indicate it need not be awaited. (the warning string explains another option is to `.catch` but that seems silly here)

File: packages/notifier/test/test-notifier-adaptor.js
Comment (turadg) [reply packages/notifier/test/test-notifier-adaptor.js]: Makes sense. I was wrong about the parameterization

File: packages/notifier/src/notifier.js
Comment (turadg) [reply packages/notifier/src/notifier.js]: I've also seen `P` for Presence, kind of the opposite.

I don't think we have any consistent guidelines on this but I would make the case that when possible we rely on the type system for whether something is a promise or not. If there's no need to bind both the promise and the resolved value then use the simpler name. non-blocking request.

**Review by turadg — COMMENTED** [review]
LG. I'll approve when the three new eslint warnings are resolved.

**Comment by gibson042** [comment]
I didn't see the promise ones locally. Anyway, resolved and pushed.

**Review by turadg — APPROVED** [review]
👍

File: packages/notifier/src/notifier.js
Comment (turadg) [inline packages/notifier/src/notifier.js]: This isn't an async function:
```suggestion
export const makeNotifierFromSubscriber = subscriberP => {
```

I don't think it needs to return `Promise` either. Can't it return the `Notifier` as `makeNotifierFromAsyncIterable` does?

**Review by turadg — CHANGES_REQUESTED** [review]

**Comment by turadg** [comment]
Working on https://github.com/Agoric/agoric-sdk/pull/5704/ I ran into some issues using this so I took another look and found a few things to address:
- whether it [needs to be async](https://github.com/Agoric/agoric-sdk/pull/5737#discussion_r917183858)
- [documentation](https://github.com/Agoric/documentation) ticket for notifiers.html
- some confidence that the the `???` in https://github.com/Agoric/agoric-sdk/pull/5704 aren't due to bugs here

File: packages/notifier/src/notifier.js
Comment (turadg) [reply packages/notifier/src/notifier.js]: The branch will be dead but https://github.com/Agoric/agoric-sdk/pull/5695 will endure for anyone who wants to see it.

Non-blocking but I do think the PR suffices as a record and to the point of discover as to why this is the design, I don't think commit history suffices. If that's the goal is should be written into HEAD.

**Review by turadg — APPROVED** [review]
🎉 I rebased my branch onto this version and the contract tests pass!

 :shipit: 😸

**Pull request merged**
